File: particle-auth-core/android/src/main/kotlin/com/particleauthcore/module/ReactErrorMessage.java
package com.particleauthcore.module;

import com.google.gson.annotations.SerializedName;

public class ReactErrorMessage {
    public static final int ExceptionCode = 10000;
    @SerializedName("code")
    public int code;

    @SerializedName("message")
    public String message;


    public ReactErrorMessage(int code, String message) {
        this.code = code;
        this.message = message;
    }

    public ReactErrorMessage() {
    }

    public static ReactErrorMessage exceptionMsg(String message) {
        return new ReactErrorMessage(ExceptionCode, message);
    }


}

File: particle-auth-core/android/src/main/kotlin/com/particleauthcore/module/ReactErrorMessage.kt
package com.particleauthcore.module

import androidx.annotation.Keep
import com.google.gson.annotations.SerializedName

@Keep
class ReactErrorMessage {
    @SerializedName("code")
    var code = 0

    @SerializedName("message")
    var message: String? = null

    constructor(code: Int, message: String?) {
        this.code = code
        this.message = message
    }

    constructor()

    companion object {
        const val ExceptionCode = 10000
        fun exceptionMsg(message: String?): ReactErrorMessage {
            return ReactErrorMessage(ExceptionCode, message)
        }
    }
}
